nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-service-api/src/main/java/org/apache/nifi/serialization/RecordSetWriterFactory.java

```diff
@@ -76,5 +78,23 @@
      * @return a RecordSetWriter that can write record sets to an OutputStream
      * @throws IOException if unable to read from the given InputStream
      */
-    RecordSetWriter createWriter(ComponentLog logger, RecordSchema schema, OutputStream out) throws SchemaNotFoundException, IOException;
```

Comment:
Instead of changing the interface method, wouldn't it be best to add a new method? I know we already did this kind of change in the past but it's going to break custom processors that would use the Record API. If we do that, could be good to keep it in mind and mention it in the release/upgrade note.

Reply:
I agree with @pvillard31 on this point. We can't really change service interfaces with a very compelling reason otherwise we'll alienate third party developers.